demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php

```diff
@@ -13,120 +13,154 @@
+     *
      * @return FunctionInterface<bool>
      */
     public function isAuthorizedViaPlanningAgency(): FunctionInterface
     {
-        $procedurePlanningOffices = $this->procedure->getPlanningOfficesIds();
+        if ($this->userOrProcedure instanceof User) {
+            $user = $this->userOrProcedure;
+            $organisationId = $user->getOrganisationId();
 
-        return $this->conditionFactory->propertyHasAnyOfValues(
-            $procedurePlanningOffices,
-            ['orga', 'id']
-        );
+            return $this->conditionFactory->propertyHasStringAsMember($organisationId, ['planningOffices']);
+        }
+
+        $procedure = $this->userOrProcedure;
+        $procedurePlanningOffices = $procedure->getPlanningOfficesIds();
+
+        return $this->conditionFactory->propertyHasAnyOfValues($procedurePlanningOffices, ['orga', 'id']);
     }
 
     /**
-     * If {@link GlobalConfigInterface::hasProcedureUserRestrictedAccess} is set to `false`
+     * If {@link GlobalConfigInterface::hasProcedureUserRestrictedAccess} is set to `false`,
      * then the user must be in the organisation that created the procedure.
      *
-     * If {@link GlobalConfigInterface::hasProcedureUserRestrictedAccess} is set to `true`
-     * then the user must either be authorized {@link OwnsProcedureConditionFactory::isAuthorizedViaPlanningAgency()
-     * via their planning agency} or manually **regardless of their role**.
+     * If {@link GlobalConfigInterface::hasProcedureUserRestrictedAccess} is set to `true`,
+     * then the user must be authorized manually for the procedure.
+     *
+     * The returned condition will not apply role checks by itself. Use in conjunction with
+     * {@link self::hasProcedureAccessingRole}.
      *
      * @return FunctionInterface<bool>
      */
     public function isAuthorizedViaOrgaOrManually(): FunctionInterface
     {
-        $orgaOwnsProcedure = $this->conditionFactory->propertyHasValue(
-            $this->procedure->getOrgaId(),
-            ['orga', 'id']
-        );
-
-        // T8427: allow access by manually configured users, overwriting the organisation-based access
-        if ($this->globalConfig->hasProcedureUserRestrictedAccess()) {
-            $planningAgencyIsAuthorized = $this->isAuthorizedViaPlanningAgency();
-            $userIsAuthorized = $this->conditionFactory->propertyHasAnyOfValues(
-                $this->procedure->getAuthorizedUserIds(),
-                ['id']
-            );
-
-            $orgaOwnsProcedure = $this->conditionFactory->anyConditionApplies(
-                $userIsAuthorized,
-                $planningAgencyIsAuthorized
-            );
-        }
-
-        return $orgaOwnsProcedure;
+        // T8427: allow access by manually configured users if the config is set to `true`,
+        // overwriting the organisation-based access
+        return $this->globalConfig->hasProcedureUserRestrictedAccess()
+            ? $this->userIsExplicitlyAuthorized()
```

Comment:
Isnt in the true case here now the permissions for planningAgencies ditched? Now only authorized users are checked, not anymore whether planning agency is permitted to procedure, if I see it correctly

Reply:
Yes that was one of the weird inconsistencies that was noticed some time (about a year?) ago but not dared to be touched. I tried to confirm the intended behavior with Andrea via Slack. The "new" behavior (it was already implemented in a different place) will still authorize users via their planning agency (`$this->isAuthorizedViaPlanningAgency();`), but
1. to be authorized, users must have the `PRIVATE_PLANNING_AGENCY` role
2. the authorization via planning agency is active, regardless of the `hasProcedureUserRestrictedAccess` global config setting.

Again this is not really new, but consolidated behavior and so far makes sense to me logically.

How planning agency are authorized can be found by searching for the calls of `isAuthorizedViaPlanningAgency`.